sources/en.batcave/src/lib.rs

```diff
@@ -249,29 +388,51 @@ impl Home for BatCave {
 
 						let url = element.attr("abs:href");
 
-						Some(Manga {
+						Manga {
 							key: url.clone().unwrap_or_default(),
 							cover,
 							title,
 							url,
 							..Default::default()
-						})
+						}
 					})
 					.map(Into::into)
 					.collect::<Vec<Link>>()
 			})
 			.unwrap_or_default();
 
-		Ok(HomeLayout {
-			components: vec![HomeComponent {
-				title,
-				subtitle: None,
-				value: aidoku::HomeComponentValue::Scroller {
-					entries: hot_releases,
-					listing: None,
-				},
-			}],
+		Some(HomeComponent {
+			title,
+			value: aidoku::HomeComponentValue::MangaList {
+				ranking: false,
+				entries,
+				listing: None,
+				page_size: None,
+			},
+			..Default::default()
 		})
+	})
+}
+
+impl Home for BatCave {
+	fn get_home(&self) -> Result<HomeLayout> {
+		let html = Request::get(BASE_URL)?.html()?;
+		let mut components = Vec::new();
+
+		let component_fns: Vec<ComponentBuilderFn> = vec![
```

Comment:
I don't think you need an allocated vector here, you should be able to get away with just a slice.
```suggestion
		let component_fns: &[ComponentBuilderFn; 4] = &[
```